pkg/kubectl/cmd/run.go

```diff
@@ -92,6 +92,7 @@ func addRunFlags(cmd *cobra.Command) {
 	cmd.Flags().String("image", "", "The image for the container to run.")
 	cmd.MarkFlagRequired("image")
 	cmd.Flags().IntP("replicas", "r", 1, "Number of replicas to create for this container. Default is 1.")
+	cmd.Flags().Bool("rm", false, "If true, delete the object created in this command when interactive mode ended.")
```

Comment:
> docker run --rm is effective whatever the mode is, we should discuss whether and how to support that firstly in the issue #16672

Based on the doc, https://docs.docker.com/engine/reference/run/#detached-d, you can't use `--rm` in combination with `-d` (detached).